### DISCUSSION TITLE
Dynamic Vertex Problem

### DISCUSSION THREAD

**Discussion by xtl-murphy** [google/filament#3683] [Q&A] [answered]
Hi Filament community,

I'm trying to change my vertices data like add/remove，but the vbo is STATIC.

if my vertices size more than the vertex size, how should i do.

**Comment by prideout** [comment, marked as answer]
You can't change the VBO size, but you can change the active range of indices in the index buffer by calling `setGeometryAt`.    This is useful if you know the maximum size ahead of time.

Alternatively you can simply create a brand new VertexBuffer when your required size changes beyond the initial capacity.  For an example of this, check out `ImGuiHelper.cpp`.  It grows the buffer by 2x each time by recreating the VertexBuffer.

**Reply by xtl-murphy** [reply]
thank you
